src/features/collapsible-sidebar/CollapsibleSidebarMenuItem.js

```diff
@@ -71,33 +78,101 @@ const StyledMenuItemLabel = styled.span`
     }
 `;
 
+// {...rest} props will go here, such as `as` prop to adjust the component name.
+// In most cases the consumer will want the tag to use a `Link` instead of `a`.
+const StyledLink = styled.a`
+    display: flex;
+    align-items: center;
+    height: ${({ theme }) => theme.base.gridUnitPx * 10}px;
+    padding: 0 ${({ theme }) => theme.base.gridUnitPx * 3}px;
+    overflow-x: hidden;
+    color: ${({ theme }) => theme.primary.foreground};
+    font-weight: bold;
+    white-space: nowrap;
+    border: 1px solid transparent;
+    border-radius: ${({ theme }) => theme.base.gridUnitPx * 2}px;
+    transition: background-color 0.15s cubic-bezier(0.215, 0.61, 0.355, 1);
+
+    &:hover,
+    &:active,
+    &:focus,
+    &.is-currentPage {
+        .bdl-CollapsibleSidebar-menuItemIcon,
+        .bdl-CollapsibleSidebar-menuItemLabel {
+            opacity: 1;
+        }
+    }
+
+    &:focus {
+        border-color: ${({ theme }) => theme.primary.foreground};
+        outline: none;
+    }
+
+    &:focus:active {
+        border-color: transparent;
+    }
+
+    &.is-currentPage {
+        background-color: ${({ theme }) => theme.primary.backgroundActive};
+    }
+
+    &.is-currentPage:active {
+        border-color: ${({ theme }) => theme.primary.foreground};
+    }
+
+    .bdl-CollapsibleSidebar-menuItemIcon + .bdl-CollapsibleSidebar-menuItemLabel {
+        margin-left: 16px;
+    }
+
+    &.show-action + .bdl-CollapsibleSidebar-menuItemActionContainer,
+    &:focus + .bdl-CollapsibleSidebar-menuItemActionContainer,
+    &:hover + .bdl-CollapsibleSidebar-menuItemActionContainer {
+        opacity: 1;
+    }
+`;
+
 type Props = {
     /** Additional classes */
     className?: string,
     /** Content to be  displayed next to item label. */
     content?: React.Node,
     icon?: React.Node,
     overflowAction?: React.Node,
+    /** Default is to always show */
+    showAction?: 'hover' | 'always',
```

Comment:
Nit: Should this maybe be `showOverflowActionOn` or something like that?

Reply:
Done